### DISCUSSION TITLE
status from Mutation not set to "success" when using onSuccess

### DISCUSSION THREAD

**Discussion by megleighc** [TanStack/query#2928] [Q&A]
Hi all,

I've done my best to read through the documentation and blog posts so I may have missed something to explain this.

I have a useMutation hook that, if I comment out the onSuccess func in the hook, will update the `status` returned from the mutation in the component to "success" (and therefore updates isSuccess to true). However, if I add functionality into the onSuccess the status never gets set to true and I'm not able to render something I'm wanting in my component when the mutation succeeds.

my mutation hook:
`const useUpdateItem = () => {
return useMutation(agreeToUpdate, {
onSuccess: (data) => {
// do stuff
}
})
}`

and in my component I'm calling the hook and pulling out isSuccess:
`const { isSuccess } = useUpdateItem();
isSuccess && (<SomeComponent />)`
If I log `status` in the component it shows "loading" and then "idle", but never "success".

Am I not supposed to be using isSuccess in my component if I use onSuccess in the mutation?

Apologies if this has already been asked, I did try to find any existing similar questions.

**Comment by TkDodo** [comment]
This sounds like it shouldn’t happen. Can you reproduce this in a codesandbox please?

**Reply by megleighc** [reply]
I can't for the life of me reproduce this in a sandbox with generic example so it must be something specific in my code. No idea what, but hopefully I'll figure it out 😓
